exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetIsPredicate.java

```diff
@@ -124,8 +124,7 @@ private static LogicalExpression createIsTruePredicate(LogicalExpression expr) {
    */
   private static LogicalExpression createIsFalsePredicate(LogicalExpression expr) {
     return new ParquetIsPredicate<Boolean>(expr, (exprStat, evaluator) ->
-        //if min value is not false or if there are all nulls  -> canDrop
-        isAllNulls(exprStat, evaluator.getRowCount()) || exprStat.hasNonNullValue() && ((BooleanStatistics) exprStat).getMin()
+      exprStat.hasNonNullValue() && ((BooleanStatistics) exprStat).getMin() || isAllNulls(exprStat, evaluator.getRowCount()) ? RowsMatch.NONE : checkNull(exprStat)
```

Comment:
Under what condition `hasNonNullValue() && isAllNulls()` will be `true`? What unit test covers this use case?

Reply:
hasNonNullValue = true if min and max exist
isAllNulls = true if all rows are null values
testBooleanPredicate with File 0_0_3.parquet (contains only 3 null values)

Reply:
@jbimbert
- If all rows are null, what are the values for min and max, should not `hasNonNullValue` be false?
- Please point me to the specific test that validates that condition.
- I would prefer to see a unit test, not an integration test. For this particular case, the integration test validates results of a query, but it does not validate what is the result of `((ParquetFilterPredicate<Boolean>)createIsFalsePredicate(expr)).canDrop(evaluator)` is.

Reply:
@vrozov this is valid case, the same issue if described in DRILL-6603, we missed `hasNonNullValue` check for `is null` predicate and filtered out row group that we shouldn't. 
@jbimbert `isAllNulls` is used more often in your code then in previous version. In previous version it was used only three times and each time had `hasNonNullValue` check. I am afraid in your code checks without `hasNonNullValue` may produce incorrect results. I am thinking what if we modify `isAllNulls` method to include `hasNonNullValue` check inside?

Reply:
@jbimbert to answer this question you can analyze `org.apache.parquet.column.statistics.Statistics` and see what can or cannot be done.

Reply:
@jbimbert @arina-ielchiieva OK, I miscounted parentheses in the modified expression.
@jbimbert What is the reason you changed the order of `isAllNulls()` and `getMin()`? Unless there is a reason, please keep the original order.

What unit test covers the following cases: (min:false, max:false, all nulls:false) and (min:false, max:true, all nulls:false)? What is the expected output in the first case and the second case?

Reply:
@vrozov Yes, to have the same code in all lines. Easier to see maintain.

Reply:
@jbimbert 
- The original code was also the same, no need to change the order.
- Please check the use cases I provided.